### ISSUE TITLE
bug(transforms): MixtureGaussianCDF/MixtureLogisticCDF .inverse has zero gradient (unrolled bisection, no implicit diff)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#111]
## Summary

`MixtureGaussianCDF.inverse` / `MixtureLogisticCDF.inverse` (and their
`inverse_and_log_det`) return the correct **value** but a **zero gradient**.
`gauss_flows._src.utils.bisection_inverse` is an unrolled `jax.lax.fori_loop`
whose bracket updates are `jnp.where(f_mid < y, …)` comparisons, with no
`custom_vjp` / implicit-diff rule. Because the output depends on the inputs only
through those piecewise-constant comparisons, `jax.grad` through `.inverse` is
exactly `0` w.r.t. both the target `y` and the bijector parameters.

The **forward** (`transform` / `log_prob`) is closed-form and differentiates
fine, so density / MLE training is unaffected. The gap only bites when you
backprop through `sample` / `inverse` — e.g. IAF-style or reparameterized-sampling
objectives, or `jax.grad` of anything that calls `.inverse`.

## Repro

```python
import jax, jax.numpy as jnp, numpy as np
import gauss_flows as gf
jax.config.update("jax_enable_x64", True)

data = jnp.asarray(np.random.default_rng(0).normal(0.0, 1.5, size=(2000, 1)))
b = gf.MixtureGaussianCDF.from_data(data, n_components=6)
y = jnp.array([0.7])

g_inv = jax.grad(lambda yy: b.inverse(yy).sum())(y)[0]          # -> 0.0  (wrong)
xstar = b.inverse(y)
g_ok  = 1.0 / jax.grad(lambda xx: b.transform(xx).sum())(xstar)[0]  # correct: 1/F'(x*)
print(float(g_inv), float(g_ok))   # 0.0   1.1622...
```

Same for `MixtureLogisticCDF` (also routes through `bisection_inverse`).

## Expected

`grad` through `.inverse` should equal the implicit-function-theorem value:
`d x*/dy = 1 / F'(x*)` and `d x*/dθ = -∂_θF(x*) / F'(x*)`.

## Fix options

1. **optimistix** (already a dependency): replace the hand-rolled `fori_loop`
   with `optimistix.root_find(..., adjoint=ImplicitAdjoint())` — gives the exact
   gradient at O(1) memory, automatically.
2. **custom_vjp** on `bisection_inverse` implementing the implicit-function
   theorem (`cotangent_x = cotangent_out / F'(x*)`, propagate to params).
3. **one-step trick** (cheapest, ~3 lines): detach the solved root and append a
   single Newton step,
   `x* = stop_gradient(solve(y, θ)); return x* - (F(x*, θ) - y) / F'(x*, θ)`.
   Verified to recover the exact implicit gradient.

## Context

Found while writing a "mixture-CDF as a learnable bijector" tutorial that
contrasts unrolling vs. one-step vs. adjoint differentiation of the inverse.
